relayer/chain/relaychain/connection.go

```diff
@@ -233,3 +209,26 @@ func (co *Connection) FetchFinalizedParaHead(relayBlockhash types.Hash, paraID u
 
 	return &header, nil
 }
+
+func (co *Connection) FetchMMRLeafCount(relayBlockhash types.Hash) (uint64, error) {
+	mmrLeafCountKey, err := types.CreateStorageKey(co.Metadata(), "Mmr", "NumberOfLeaves", nil, nil)
+	if err != nil {
+		return 0, err
+	}
+	var mmrLeafCount uint64
+
+	ok, err := co.API().RPC.State.GetStorage(mmrLeafCountKey, &mmrLeafCount, relayBlockhash)
+	if err != nil {
+		return 0, err
+	}
+
+	if !ok || mmrLeafCount == 0 {
```

Comment:
There's actually two distinct unrelated error cases here, so I would not lump them in the with same log message as it could cause confusion. There's "MMR Leaf Count Not Found" and "MMR is empty and has no leaves"

Reply:
fixed 06a2d8d